src/uct/cuda/base/cuda_md.c

```diff
@@ -55,6 +56,60 @@ UCS_PROFILE_FUNC(ucs_status_t, uct_cuda_base_detect_memory_type,
     return UCS_ERR_INVALID_ADDR;
 }
 
+static ucs_status_t uct_cuda_base_get_sys_dev(ucs_sys_device_t *sys_dev_p)
+{
+    CUdevice dev;
+    int attrib;
+
+    if (UCS_OK != UCT_CUDADRV_FUNC(cuCtxGetDevice(&dev))) {
+        return UCS_ERR_IO_ERROR;
+    }
+
+    if (UCS_OK != UCT_CUDADRV_FUNC(cuDeviceGetAttribute(&attrib,
+                    CU_DEVICE_ATTRIBUTE_PCI_DOMAIN_ID, dev))) {
+        return UCS_ERR_IO_ERROR;
+    }
+    sys_dev_p->bus_id.domain = (uint16_t)attrib;
+
+    if (UCS_OK != UCT_CUDADRV_FUNC(cuDeviceGetAttribute(&attrib,
+                    CU_DEVICE_ATTRIBUTE_PCI_BUS_ID, dev))) {
+        return UCS_ERR_IO_ERROR;
+    }
+    sys_dev_p->bus_id.bus = (uint8_t)attrib;
+
+    return UCS_OK;
+
+}
+
+ucs_status_t uct_cuda_base_mem_query(uct_md_h md, const void *addr,
+                                     size_t length,
+                                     uct_md_mem_attr_t *mem_attr_p)
+{
+    ucs_status_t status;
+
+    mem_attr_p->field_mask = 0;
+
+    if (!addr || !length) {
+        return UCS_ERR_INVALID_PARAM;
+    }
+
+    status = uct_cuda_base_detect_memory_type(md, addr, length, &(mem_attr_p->mem_type));
+    if (UCS_OK == status) {
+        mem_attr_p->field_mask |= UCT_MD_MEM_ATTR_FIELD_MEM_TYPE;
+
+        status = uct_cuda_base_get_sys_dev(&(mem_attr_p->sys_dev));
```

Comment:
maybe we should return mem type only when it failed to detect sysdev with warning

Reply:
Agreed. I think even if didn't detect either `mem_type` or `sys_dev`, we should not return an error because the way that both these fields are consumed is by checking if `field_mask` bits are set. As long as there isn't a genuine error, then returning `field_mask = 0` and not populating `mem_type` and `sys_dev` should be valid. Thoughts on this @bureddy ?